programs/ssl/ssl_server2.c

```diff
@@ -1572,13 +1572,15 @@ int main( int argc, char *argv[] )
         else if( strcmp( p, "buffer_size" ) == 0 )
         {
             opt.buffer_size = atoi( q );
-            if( opt.buffer_size < 1 || opt.buffer_size > MBEDTLS_SSL_MAX_CONTENT_LEN + 1 )
+            if( opt.buffer_size < 1 ||
+                    ( opt.buffer_size > MBEDTLS_SSL_IN_CONTENT_LEN + 1
+                      && opt.buffer_size > MBEDTLS_SSL_OUT_CONTENT_LEN + 1 ) )
```

Comment:
Sorry for not noticing earlier, but I'm not entirely comfortable with this `buffer_size` thing. First I was going say that it should probably be `||` rather than `&&` in the condition, and then I started wondering why we need the comparison at all. `buffer_size` is the size of a dynamically allocated application buffer that will be used to hold the incoming data and then the response. How is it a problem if the buffer is larger than what the SSL layer can send or receive in one go? It's just a waste of RAM in the application, but I don't think that needs to be an error.

I suggest removing the upper bound here and in the documentation for this option.

(By the way, I checked and we already call `mbedtls_calloc( 1, opt.buffer_size + 1 )` so that's one more `+1` than necessary. But if we remove the upper bound check here, it doesn't matter.)